src/java/picard/analysis/ChimeraUtil.java

```diff
@@ -48,7 +48,7 @@ public static boolean isChimeric(final SAMRecord rec, final int maxInsertSize, f
         return !(rec.getReadUnmappedFlag() || !rec.getReadPairedFlag() || rec.getMateUnmappedFlag()) &&  // the read pair needs to be mapped
                 (Math.abs(rec.getInferredInsertSize()) > maxInsertSize ||
                 !rec.getReferenceIndex().equals(rec.getMateReferenceIndex()) ||
-                expectedOrientations.contains(SamPairUtil.getPairOrientation(rec)) ||
+                !expectedOrientations.contains(SamPairUtil.getPairOrientation(rec)) ||
                 rec.getAttribute("SA") != null);
     }
 
```

Comment:
I would propose breaking up this big block of conditions into at least 2 or 3 smaller functions with more descriptive names. I find it hard to pick apart what's going on here with so many conditions and negations.